### PR TITLE
Kia Seltos 2021 INDI Tuning + Increased Steer Limit

### PR DESCRIPTION
**Pull request by speedking456**
**Description**
KALE was not performing with PID tuning, I've improved its performance with INDI tuning.

**Verification**
Drives a lot smoother and doesn't fight the lane planner.

**Route**
c75a59efa0ecd502|2021-04-01--15-20-16

### CODE REVIEW

**Comment by speedking456** [comment]
Ready to merge

**Comment by adeebshihadeh** [comment]
Our policy for tuning PRs is that another user has to confirm that the new tuning is better. In this case, another user would have to confirm both changes.

**Comment by speedking456** [comment]
> Our policy for tuning PRs is that another user has to confirm that the new tuning is better. In this case, another user would have to confirm both changes.

As far as I know, there is only one other Kia Seltos running OP. In such a case, would I have to get them to test it as well to have this merged?

**Comment by adeebshihadeh** [comment]
Yes, usually we require 2-3 people to confirm, but seems there's only two Seltos users.

**Comment by speedking456** [comment]
> Yes, usually we require 2-3 people to confirm, but seems there's only two Seltos users.

I did the original port, does that make a difference? I would have included more optimized tuning in my original pull request, but it was useable enough when I created it. I mostly wanted to get it added for those who have my vehicle as well.

**Comment by adeebshihadeh** [comment]
Let's ask in the HKG channel in discord for feedback. If you get no response in a couple days, we'll just merge.

**Comment by adeebshihadeh** [comment]
Any update on this? Did you ask in the HKG channel and get any feedback?

**Comment by speedking456** [comment]
> Any update on this? Did you ask in the HKG channel and get any feedback?

I'm waiting for the other Seltos user to get back to me. They've said they'll try the fork but no feedback yet.

**Comment by adeebshihadeh** [comment]
This good to merge now? If they haven't responded by now, I'm fine with merging.

**Comment by speedking456** [comment]
> This good to merge now? If they haven't responded by now, I'm fine with merging.

Good to merge

File: selfdrive/car/hyundai/interface.py [hunk 1]
```diff
@@ -144,13 +144,20 @@ def get_params(candidate, fingerprint=gen_empty_fingerprint(), car_fw=[]):  # py
...
-      ret.wheelbase = 2.6
-      ret.steerRatio = 13.73  # Spec
-      tire_stiffness_factor = 0.5
-      ret.lateralTuning.pid.kiBP, ret.lateralTuning.pid.kpBP = [[0.], [0.]]
-      ret.lateralTuning.pid.kpV, ret.lateralTuning.pid.kiV = [[0.25], [0.05]]
+      ret.mass = 1337. + STD_CARGO_KG
+      ret.wheelbase = 2.63
+      ret.steerRatio = 14.56
+      tire_stiffness_factor = 1
+      ret.steerActuatorDelay = 0.096
```
Comment (adeebshihadeh) [inline selfdrive/car/hyundai/interface.py:151]: revert this

File: selfdrive/car/hyundai/interface.py [hunk 2]
```diff
@@ -144,13 +144,20 @@ def get_params(candidate, fingerprint=gen_empty_fingerprint(), car_fw=[]):  # py
...
-      ret.mass = 1310. + STD_CARGO_KG
-      ret.wheelbase = 2.6
-      ret.steerRatio = 13.73  # Spec
-      tire_stiffness_factor = 0.5
-      ret.lateralTuning.pid.kiBP, ret.lateralTuning.pid.kpBP = [[0.], [0.]]
-      ret.lateralTuning.pid.kpV, ret.lateralTuning.pid.kiV = [[0.25], [0.05]]
+      ret.mass = 1337. + STD_CARGO_KG
+      ret.wheelbase = 2.63
+      ret.steerRatio = 14.56
+      tire_stiffness_factor = 1
```
Comment (adeebshihadeh) [inline selfdrive/car/hyundai/interface.py:150]: delete this, it's already the default

File: selfdrive/car/hyundai/interface.py [hunk 3]
```diff
@@ -144,13 +144,15 @@ def get_params(candidate, fingerprint=gen_empty_fingerprint(), car_fw=[]):  # py
       ret.lateralTuning.pid.kiBP, ret.lateralTuning.pid.kpBP = [[0.], [0.]]
       ret.lateralTuning.pid.kpV, ret.lateralTuning.pid.kiV = [[0.25], [0.05]]
     elif candidate == CAR.KIA_SELTOS:
-      ret.lateralTuning.pid.kf = 0.00005
-      ret.mass = 1310. + STD_CARGO_KG
```
Comment (adeebshihadeh) [inline selfdrive/car/hyundai/interface.py:148]: why are all these other lines deleted? only remove the ones for the PID tuning

Comment (speedking456) [reply selfdrive/car/hyundai/interface.py:148]: The PID lines were already removed, didn't get what you meant by
> delete this, it's already the default
I've reverted the changes, it should have already been correct I would think.

**Review by speedking456 — COMMENTED** [review]
I changed these params as my original pull request was not entirely accurate. The updated params are more accurate.

**Comment by adeebshihadeh** [comment]
Ah, this PR should have been split up in that case, but we can merge it in one this time. Can you update the params? Then this should be good to merge.

**Comment by speedking456** [comment]
> Ah, this PR should have been split up in that case, but we can merge it in one this time. Can you update the params? Then this should be good to merge.

Un-reverted, sorry about the confusion, didn't realize it should be two PRs. I'll keep that in mind next time.

Ready to merge whenever you are.

**Comment by adeebshihadeh** [comment]
One last thing: revert the change to `steerActuatorDelay`. It's an unnoticeable change and those values should be determined from lots of data.

**Comment by speedking456** [comment]
> One last thing: revert the change to `steerActuatorDelay`. It's an unnoticeable change and those values should be determined from lots of data.

Reverted. 

I originally made that change thinking it improved performance on two car wide roads with only a center line, however I think it was the model improvements that fixed it.

**Pull request merged**